pkg/resolvers/grbClusterResolver.go

```diff
@@ -18,18 +18,46 @@ const (
 // GRBClusterRuleResolver implements the rbacv1.AuthorizationRuleResolver interface. Provides rule resolution
 // for the permissions a GRB gives that apply in a given cluster (or all clusters).
 type GRBClusterRuleResolver struct {
-	GlobalRoleBindings v3.GlobalRoleBindingCache
-	GlobalRoleResolver *auth.GlobalRoleResolver
+	gbrCache     v3.GlobalRoleBindingCache
+	grResolver   *auth.GlobalRoleResolver
+	ruleResolver func(namespace string, gr *apisv3.GlobalRole, grResolver *auth.GlobalRoleResolver) ([]rbacv1.PolicyRule, error)
 }
 
-// New NewGRBClusterRuleResolver returns a new resolver for resolving rules given through GlobalRoleBindings
-// which apply to cluster(s). This function can only be called once for each unique instance of grbCache.
-func NewGRBClusterRuleResolver(grbCache v3.GlobalRoleBindingCache, grResolver *auth.GlobalRoleResolver) *GRBClusterRuleResolver {
+// NewGRRuleResolvers returns resolvers for resolving rules given through GlobalRoleBindings
+// which apply to cluster(s). This function can only be called once for each unique instance of GlobalRoleBindings.
+func NewGRRuleResolvers(grbCache v3.GlobalRoleBindingCache, grResolver *auth.GlobalRoleResolver) (*GRBClusterRuleResolver, *GRBClusterRuleResolver, *GRBClusterRuleResolver) {
```

Comment:
what if we aggregate the three resolvers? I aggregated them in a new struct `GRBRuleResolvers`. That way we can reference them in the validator, and it's clear which rules are being resolved. Let me know what you think

Reply:
Yep, I think that this works. Since the struct is exported, I think it would be good to add comments above each field to specify what exactly that specific resolver is for, but overall I think that this structure works. Resolving.